### PR TITLE
docs: iteration 3

### PR DESCRIPTION
**Pull request by casperdcl**
- [x] ~update README USPs~ split into features & USPs
  + [x] sync with `docs/index.md`
- [x] update description
- [x] add high-level diagram
- [x] more info on cloud account creation
- [x] update example scripts (make spot recovery use case more clear)
- [x] fix cross-references/pluralisation
- [x] add low-level diagram

---

### high-level

<details><summary>inspired by</summary>

- https://dvc.org/doc/use-cases/fast-data-caching-hub
- https://dvc.org/doc/use-cases/ci-cd-for-machine-learning
- https://registry.terraform.io/
- image from https://github.com/iterative/dvc.org/pull/3399:

</details>

### low-level
```mermaid
flowchart LR
subgraph tpi [what TPI manages]
direction LR
    subgraph you [what you manage]
        direction LR
        A([Personal Computer])
    end
    B[("Cloud Storage (low cost)")]
    C{{"Cloud instance scaler (zero cost)"}}
    D[["Cloud (spot) Instance"]]
    A ---> |create cloud storage| B
    A --> |create cloud instance scaler| C
    A ==> |upload script & workdir| B
    A -.-> |"offline (lunch break)"| A
    C -.-> |"(re)provision instance"| D
    D ==> |run script| D
    B <-.-> |persistent workdir cache| D
    D ==> |script end,\nshutdown instance| B
    D -.-> |outage| C
    B ==> |download output| A
end
style you fill:#FFFFFF00,stroke:#13ADC7
style tpi fill:#FFFFFF00,stroke:#FFFFFF00,stroke-width:0px
style A fill:#13ADC7,stroke:#333333,color:#000000
style B fill:#945DD5,stroke:#333333,color:#000000
style D fill:#F46737,stroke:#333333,color:#000000
style C fill:#7B61FF,stroke:#333333,color:#000000
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,19 +8,21 @@
 
 TPI is a [Terraform](https://terraform.io) plugin built with machine learning in mind. Full lifecycle management of computing resources (including GPUs and respawning spot instances) from several cloud vendors (AWS, Azure, GCP, K8s)... without needing to be a cloud expert.
 
-- **Provision Resources**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
-- **Sync & Execute**: easily sync & run local data & code in the cloud
+- **Easy to use**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
 - **Low cost**: transparent auto-recovery from interrupted low-cost spot/preemptible instances
+- **No cloud vendor lock-in**: switch between several cloud vendors with ease due to concise unified configuration
+- **Seamless developer experience**: easily sync & run data & code in the cloud as easily as on a local laptop
```
[the remark this answers is not in the document]
Comment (DavidGOrtega) [reply README.md:14]: **Open source: No proprietary locks and you are able to extend it

**Review by DavidGOrtega — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -119,6 +130,38 @@ Instead of using the latest stable release, a local copy of the repository must
    ```
 3. Use `source = "github.com/iterative/iterative"` in your `main.tf` to use the local repository (`source = "iterative/iterative"` will download the latest release instead), and run `terraform init --upgrade`
 
+This diagram may also help to see what TPI does under-the-hood:
```
Comment (DavidGOrtega) [inline README.md:133]:
```suggestion
## How does it work

This diagram may also help to see what TPI does under-the-hood:
```

Comment (DavidGOrtega) [reply README.md:133]: Prior to ## Contributing

**Review by DavidGOrtega — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -8,19 +8,21 @@
 
 TPI is a [Terraform](https://terraform.io) plugin built with machine learning in mind. Full lifecycle management of computing resources (including GPUs and respawning spot instances) from several cloud vendors (AWS, Azure, GCP, K8s)... without needing to be a cloud expert.
 
-- **Provision Resources**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
-- **Sync & Execute**: easily sync & run local data & code in the cloud
+- **Easy to use**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md:11]: good question.

- What are target audiences for the docs?
- What precedence do said audiences take?

For release week, I was thinking mostly driven by views :eyes: rather than bookmarks :star:

1. newbies: no cloud account (10k+ :eyes:)
2. mlops & devops: have cloud accounts and hate setup time (1k+ :eyes:)
3. ml engineers & data scientists: a mix of (1) & (2), don't need separate category

File: README.md [hunk 4]
```diff
@@ -119,6 +130,38 @@ Instead of using the latest stable release, a local copy of the repository must
...
+    A -.-> |"offline (lunch break)"| A
+    C -.-> |"(re)provision instance"| D
+    D ==> |run script| D
+    B <-.-> |persistent workdir cache| D
+    D ==> |script end,\nshutdown instance| B
+    D -.-> |outage| C
+    B ==> |download output| A
+end
+style you fill:#fff,stroke:#13ADC7
+style tpi fill:#fff,stroke:#fff,stroke-width:0px
```
Comment (casperdcl) [inline README.md:158]:
```suggestion
style you fill:#FFFFFF,stroke:#13ADC7
style tpi fill:#FFFFFF,stroke:#FFFFFF,stroke-width:0px
```

File: README.md [hunk 5]
```diff
@@ -119,6 +130,38 @@ Instead of using the latest stable release, a local copy of the repository must
...
+style A fill:#13ADC7,stroke:#333
+style B fill:#945DD5,stroke:#333
+style D fill:#F46737,stroke:#333
+style C fill:#7B61FF,stroke:#333
```
Comment (casperdcl) [inline README.md:162]:
```suggestion
style you fill:#FFFFFF00,stroke:#13ADC7
style tpi fill:#FFFFFF00,stroke:#FFFFFF00,stroke-width:0px
style A fill:#13ADC7,stroke:#333333,color:#000000
style B fill:#945DD5,stroke:#333333,color:#000000
style D fill:#F46737,stroke:#333333,color:#000000
style C fill:#7B61FF,stroke:#333333,color:#000000
```

**Comment by jendefig** [comment]
Question:  Where in the Readme will the new banner image go?  I see where the low-level diagram goes, but not the banner.  It's not replacing the Terraform/Iterative banner, right?

**Comment by casperdcl** [comment]
> Where in the Readme will the new [high-level] banner image go?

I was thinking somewhere mid-way. Potentially just below the USP bullet list.

File: README.md [hunk 6]
```diff
@@ -8,19 +8,21 @@
...
-Supported cloud vendors include:
+Supported cloud vendors [include][auth]:
 
 - Amazon Web Services (AWS)
 - Microsoft Azure
 - Google Cloud Platform (GCP)
 - Kubernetes (K8s)
 
+[auth]: https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication
+
```
Comment (casperdcl) [inline README.md:25]: TODO: insert high-level daigram here

```suggestion

![](todo.png)

```

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmpetrov) [reply README.md:11]: @DavidGOrtega what would be your recommendation for the 1st one?

**Review by dmpetrov — CHANGES_REQUESTED** [review]
It feels like we are lacking the `unique competitive advantage` or `Why TPI?`. I there a way to introduce it?

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (DavidGOrtega) [reply README.md:11]: The first issue that I find is logically, when I read it It sounds to me that we are offering the `machine` and not the `task`. 
Aside that the real problem to me is that "easy" is nice but why should I change my current aws thingy to use this. 
To me the principal motivation to do so is to know that this `task` allows me to do heavy ops in spot instances hassle-free so I would focus on what the task.
I would not put it in the first place and more after `No cloud vendor lock-in` since I would have introduced the clouds  and I would slightly rephrase it to:

**Easy to use**: Launch a task without infrastructure knowledge or having to be a Devops expert, anyone can do it.

File: README.md [hunk 7]
```diff
@@ -8,19 +8,21 @@
 
 TPI is a [Terraform](https://terraform.io) plugin built with machine learning in mind. Full lifecycle management of computing resources (including GPUs and respawning spot instances) from several cloud vendors (AWS, Azure, GCP, K8s)... without needing to be a cloud expert.
 
-- **Provision Resources**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
-- **Sync & Execute**: easily sync & run local data & code in the cloud
+- **Easy to use**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
 - **Low cost**: transparent auto-recovery from interrupted low-cost spot/preemptible instances
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md:12]:
```suggestion
- **Lower cost**: transparent auto-recovery from interrupted low-cost spot/preemptible instances
```

File: README.md [hunk 8]
```diff
@@ -8,19 +8,21 @@
...
+- **No cloud vendor lock-in**: switch between several cloud vendors with ease due to concise unified configuration
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md:13]:
```suggestion
- **No cloud vendor lock-in**: our unified abstraction allows switching between clouds with just one line
```

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md:14]: - dev experience: "cloud vs laptop" seems better than "CI vs laptop"
- open source: yes, but not really a USP (emphasis on "U")? Maybe could be last on the list.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md:11]: current order:

- Easy to use - newbie friendly
- Lower cost - expert friendly
- No cloud vendor lock-in - expert friendly
- Seamless developer experience - neutral
- No waste - expert friendly

is this your proposed order @DavidGOrtega?

- Lower cost - expert friendly
- Easy to use - newbie friendly
- No cloud vendor lock-in - expert friendly
- Seamless developer experience - neutral
- No waste - expert friendly

[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply README.md:11]: > _expert friendly_

What somebody calls “FinOps” I guess... 😅

File: docs/guides/authentication.md [hunk 9]
```diff
@@ -48,6 +52,8 @@ export AZURE_CLIENT_SECRET="$(terraform output --raw azure_client_secret)"
 
 ## Google Cloud Platform
 
+[Create a GCP account](https://cloud.google.com/free) if needed, and then set the environment variable:
+
 - `GOOGLE_APPLICATION_CREDENTIALS` - Path to (or contents of) a service account JSON key file.
```
Comment (dacbd) [inline docs/guides/authentication.md:57]:
```suggestion
Either one of:
- `GOOGLE_APPLICATION_CREDENTIALS` - Path to a service account JSON key file.
- `GOOGLE_APPLICATION_CREDENTIALS_DATA` - Contents of a service account JSON key file
```

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md]: reworked this with a feature-USP split; please re-review :pray:

File: README.md [hunk 10]
```diff
@@ -59,8 +61,17 @@ resource "iterative_task" "example" {
...
-    mkdir results
-    echo "Hello World!" > results/greeting.txt
+
+    # create output directory if needed
+    mkdir -p results
+    # read last result (in case of spot/preemptible instance recovery)
+    if [[ -f results/epoch.txt ]]; then EPOCH="$(cat results/epoch.txt)"; fi
+
+    # (re)start training loop up to 42 epochs
+    for epoch in $(seq ${EPOCH:-1} 42); do
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md:70]: yes we'll definitely add the 3 pending blogs when they're merged (https://github.com/iterative/dvc.org/pull/3399, https://github.com/iterative/dvc.org/pull/3087, https://github.com/iterative/dvc.org/pull/3246)

File: docs/guides/authentication.md (hunk 9 above)
Comment (casperdcl) [reply docs/guides/authentication.md:57]: /CC @0x2b3bfa0 on this one :)

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md]: reworded & split, worth a review :pray:

File: README.md [hunk 11]
```diff
@@ -8,19 +8,22 @@
...
-- **No lock-in**: switch between several cloud vendors with ease due to concise unified configuration
+- **Seamless developer experience**: easily sync data & run code with one command, making the cloud feel like a laptop
 
-Supported cloud vendors include:
+Supported cloud vendors [include][auth]:
 
 - Amazon Web Services (AWS)
 - Microsoft Azure
 - Google Cloud Platform (GCP)
 - Kubernetes (K8s)
```
Comment (casperdcl) [inline README.md:21]: technically should fit on one line:

```suggestion
Amazon Web Services (AWS) | Microsoft Azure | Google Cloud Platform (GCP) | Kubernetes (K8s)
```

File: docs/guides/authentication.md (hunk 9 above)
Comment (0x2b3bfa0) [reply docs/guides/authentication.md:57]:
```suggestion
Either one of:
- `GOOGLE_APPLICATION_CREDENTIALS` - **Path to** a service account JSON key file
- `GOOGLE_APPLICATION_CREDENTIALS_DATA` - **Contents of** a service account JSON key file
```

Comment (0x2b3bfa0) [reply docs/guides/authentication.md:57]: 🔔 @dacbd & @casperdcl, better now?

File: README.md (hunk 11 above)
Comment (0x2b3bfa0) [reply README.md]: Yes, sure, so nobody notices that Microsoft Azure doesn't have an official acronym[^1]

[^1]: Although the command–line tool is `az`

**Review by DavidGOrtega — CHANGES_REQUESTED** [review]
Great iteration! This is looking better and better!!

My feedback:
 -  [stop tasks](https://github.com/iterative/terraform-provider-iterative/tree/docs-iter#stop-tasks) We are not stoping tasks, we are destroying tasks, we do not have a stop/pause mechanism, if the user do this she will not be able to resume
 - `Why TPI?` is the same as the introduction and we are introduction an issue also here. Features A, B, C, D should be like a song we must enforce the user to be able to enumerate them by them selves, we are here explaining sometimes it better than in the intro but we are reprasing the feature. `Lower cost with spot recovery` -> `Spot auto-recovery`

Same happens with index.md that we are again rephrasing them and altering the order.
We have to avoid this.

File: README.md [hunk 12]
```diff
@@ -59,8 +60,17 @@ resource "iterative_task" "example" {
   }
   script = <<-END
     #!/bin/bash
-    mkdir results
-    echo "Hello World!" > results/greeting.txt
+
+    # create output directory if needed
+    mkdir -p results
+    # read last result (in case of spot/preemptible instance recovery)
+    if [[ -f results/epoch.txt ]]; then EPOCH="$(cat results/epoch.txt)"; fi
```
Comment (0x2b3bfa0) [inline README.md:67]:
```suggestion
    if test -f results/epoch.txt; then EPOCH="$(cat results/epoch.txt)"; fi
```

POSIX–friendly?

**Review by dmpetrov — APPROVED** [review]
Great change. It's good to merge after @casperdcl incorporates the comparative advantages.

File: docs/guides/authentication.md (hunk 9 above)
Comment (casperdcl) [reply docs/guides/authentication.md:57]: iirc @0x2b3bfa0 we wanted to soft-deprecate `GOOGLE_APPLICATION_CREDENTIALS_DATA`? Not sure...

Comment (0x2b3bfa0) [reply docs/guides/authentication.md:57]: I don't recall so. What we wanted to deprecate is `KUBERNETES_CONFIGURATION` in favor of the most consistent `KUBECONFIG_DATA` I think.

Comment (0x2b3bfa0) [reply docs/guides/authentication.md:57]: Google Cloud and Kubernetes only take a path, and that's why we offer these unofficial alternative environment variables, so users don't have to expose secrets as files.

File: README.md (hunk 12 above)
Comment (casperdcl) [reply README.md:82]: I counter with https://github.com/iterative/terraform-provider-iterative/blob/docs-iter/task/common/machine/script.go#L112

Comment (casperdcl) [reply README.md]: 1bb47a3 be happy :)

Comment (0x2b3bfa0) [reply README.md]: 🐲🔥✈️

Comment (0x2b3bfa0) [reply README.md]: If it has to be homogeneous, I don't know if it's better to use `[[` here. I tend to use `test` with `-options` and `[[` with `==` comparison operators. 🙃

File: README.md [hunk 13]
```diff
@@ -8,18 +8,18 @@
 
 TPI is a [Terraform](https://terraform.io) plugin built with machine learning in mind. Full lifecycle management of computing resources (including GPUs and respawning spot instances) from several cloud vendors (AWS, Azure, GCP, K8s)... without needing to be a cloud expert.
 
-- **Provision Resources**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
-- **Sync & Execute**: easily sync & run local data & code in the cloud
-- **Low cost**: transparent auto-recovery from interrupted low-cost spot/preemptible instances
+- **Lower cost with spot recovery**: transparent auto-recovery from interrupted low-cost spot/preemptible instances
```
[the remark this answers is not in the document]
Comment (casperdcl) [reply README.md:11]: made a lot of changes since this review, a couple outstanding things:

> We are not stoping tasks, we are destroying tasks

technically we *are* stopping tasks <- terminating instances <- destroying resources

> automatic intermediate data backup

implied by transparent spot auto-recovery, right?

File: README.md [hunk 14]
```diff
@@ -92,14 +126,52 @@ TF_LOG_PROVIDER=INFO terraform refresh
...
+    A ---> |create cloud storage| B
+    A --> |create cloud instance scaler| C
+    A ==> |upload script & workdir| B
+    A -.-> |"offline (lunch break)"| A
+    C -.-> |"(re)provision instance"| D
+    D ==> |run script| D
+    B <-.-> |persistent workdir cache| D
+    D ==> |script end,\nshutdown instance| B
+    D -.-> |outage| C
+    B ==> |download output| A
```
Comment (casperdcl) [inline README.md:161]:
```suggestion
    A ---> |2. create cloud storage| B
    A --> |1. create cloud instance scaler| C
    A ==> |3. upload script & workdir| B
    A -.-> |"4. offline (lunch break)"| A
    C -.-> |"5. (re)provision instance"| D
    D ==> |7. run script| D
    B <-.-> |6. persistent workdir cache| D
    D ==> |8. script end,\nshutdown instance| B
    D -.-> |outage| C
    B ==> |9. download output| A
```

File: README.md [hunk 15]
```diff
@@ -8,18 +8,41 @@
...
+
+## What's Special
+
+There are a several reasons to use TPI instead of other related solutions (custom scripts and/or cloud orchestrators):
+
+1. **Reduce management overhead and infrastructure cost**:
+   TPI is a CLI tool, not a running service. It requires no additional orchestrating machine (control plane/head nodes) to schedule/recover/terminate instances. Instead, TPI runs (spot) instances via cloud-native scaling groups[^scalers], taking care of recovery and termination automatically on the cloud provider's side. This design reduces management overhead & infrastructure costs. You can close your laptop while cloud tasks are running -- auto-recovery happens even if you are offline.
+2. **Unified tool for data science and software development teams**:
+   TPI provides consistent tooling for both data scientists and DevOps engineers, improving cross-team collaboration. This simplifies compute management to a single config file, and reduces time to deliver ML models into production.
+
```
Comment (casperdcl) [inline README.md:42]:
```suggestion
3. **Reproducible, codified environments**: Store hardware requirements & pipelines in a single configuration file with the rest of your ML project code.

```

File: docs/index.md [hunk 16]
```diff
@@ -1,27 +1,47 @@
...
+   TPI is a CLI tool, not a running service. It requires no additional orchestrating machine (control plane/head nodes) to schedule/recover/terminate instances. Instead, TPI runs (spot) instances via cloud-native scaling groups ([AWS Auto Scaling Groups](https://docs.aws.amazon.com/autoscaling/ec2/userguide/what-is-amazon-ec2-auto-scaling.html), [Azure VM Scale Sets](https://azure.microsoft.com/en-us/services/virtual-machine-scale-sets), [GCP managed instance groups](https://cloud.google.com/compute/docs/instance-groups#managed_instance_groups), and [Kubernetes Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/job)), taking care of recovery and termination automatically on the cloud provider's side. This design reduces management overhead & infrastructure costs. You can close your laptop while cloud tasks are running -- auto-recovery happens even if you are offline.
+2. **Unified tool for data science and software development teams**:
+   TPI provides consistent tooling for both data scientists and DevOps engineers, improving cross-team collaboration. This simplifies compute management to a single config file, and reduces time to deliver ML models into production.
+
```
Comment (casperdcl) [inline docs/index.md:40]:
```suggestion
3. **Reproducible, codified environments**: Store hardware requirements & pipelines in a single configuration file with the rest of your ML project code.

```

**Review by casperdcl — COMMENTED** [review]
potential 3rd point IaC/HaC

File: README.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (DavidGOrtega) [reply README.md:11]: > implied by transparent spot auto-recovery, right?

Yes and no, you make the users to think or guess at least. There is a phase in design (title also of a praised book) "Dont make me think"

[the remark this answers is not in the document]
Comment (DavidGOrtega) [reply README.md:11]: > technically we are stopping tasks

No, technically Tasks are stopped by themselves, what I want to say with this is that the user might find the ambiguity of the stop concept in cloud. Where they can stop the machine and then at some point they can restart the machine again. Here after destroy there is no possible resume

**Review by dmpetrov — APPROVED** [review]
Looks greet!

A couple of minor changes:
1. It might be better to have a link to CML repository and cml.dev, not a doc
2. It was a good idea about adding 3rd item to the list like "Reproducible, codified environments" or " Extend your GitOps and CI/CD-oriented workflows" or "hardware as code” for AI/ML" or multiple of these.

**Comment by 0x2b3bfa0** [comment]
🔔 @dmpetrov & @iterative/cml, we still have some unresolved conversations that are unlikely to have a noticeable influence:

* https://github.com/iterative/terraform-provider-iterative/pull/492#pullrequestreview-941806051
  * https://github.com/iterative/terraform-provider-iterative/pull/492#discussion_r850097139
* https://github.com/iterative/terraform-provider-iterative/pull/492#pullrequestreview-942818544
  * https://github.com/iterative/terraform-provider-iterative/pull/492#discussion_r850819781
* https://github.com/iterative/terraform-provider-iterative/pull/492#pullrequestreview-942882781
  * https://github.com/iterative/terraform-provider-iterative/pull/492#discussion_r850841913
  * https://github.com/iterative/terraform-provider-iterative/pull/492#discussion_r850842395

[Can we merge this in the current state](https://iterativeai.slack.com/archives/C01900GSB4J/p1650057090365349?thread_ts=1650055940.277949&cid=C01900GSB4J) and address them in a separate pull request?

File: README.md (hunk 14 above)
Comment (dmpetrov) [reply README.md:161]: Looks good.

**Comment by 0x2b3bfa0** [comment]
Merging as per [this conversation](https://iterativeai.slack.com/archives/C01900GSB4J/p1650064810464869?thread_ts=1650055940.277949&cid=C01900GSB4J)

File: README.md [hunk 17]
```diff
@@ -1,25 +1,48 @@
...
+![](https://github.com/iterative/static/raw/main/img/tpi/high-level-dark.png#gh-dark-mode-only)
+
+## What's Special
+
+There are a several reasons to use TPI instead of other related solutions (custom scripts and/or cloud orchestrators):
+
+1. **Reduced management overhead and infrastructure cost**:
+   TPI is a CLI tool, not a running service. It requires no additional orchestrating machine (control plane/head nodes) to schedule/recover/terminate instances. Instead, TPI runs (spot) instances via cloud-native scaling groups[^scalers], taking care of recovery and termination automatically on the cloud provider's side. This design reduces management overhead & infrastructure costs. You can close your laptop while cloud tasks are running -- auto-recovery happens even if you are offline.
+2. **Unified tool for data science and software development teams**:
+   TPI provides consistent tooling for both data scientists and DevOps engineers, improving cross-team collaboration. This simplifies compute management to a single config file, and reduces time to deliver ML models into production.
```
Comment (0x2b3bfa0) [inline README.md:41]: > _and reduces time to deliver ML models into production_

Is this a direct effect of using TPI for development? Given that this tool is not intended for model serving, that assertion migh be slightly misleading.

File: README.md [hunk 18]
```diff
@@ -1,25 +1,48 @@
...
 
-- **Provision Resources**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
-- **Sync & Execute**: easily sync & run local data & code in the cloud
-- **Low cost**: transparent auto-recovery from interrupted low-cost spot/preemptible instances
-- **No waste**: auto-cleanup unused resources (terminate compute instances upon job completion/failure & remove storage upon download of results)
-- **No lock-in**: switch between several cloud vendors with ease due to concise unified configuration
+- **Lower cost with spot recovery**: transparent auto-recovery from interrupted low-cost spot/preemptible instances
+- **No cloud vendor lock-in**: switch between clouds with just one line thanks to unified abstraction
+- **No waste**: auto-cleanup unused resources (terminate compute instances upon task completion/failure & remove storage upon download of results), pay only for what you use
```
Comment (jorgeorpinel) [inline README.md:13]: 💅🏼  missing `.` periods in these

File: README.md [hunk 19]
```diff
@@ -1,25 +1,48 @@
...
+- **Developer-first experience**: one-command data sync & code execution with no external server, making the cloud feel like a laptop
 
-Supported cloud vendors include:
+Supported cloud vendors [include][auth]:
```
Comment (jorgeorpinel) [inline README.md:16]: 💅🏼  I'd link the other 3 words.

File: README.md [hunk 20]
```diff
@@ -1,25 +1,48 @@
...
+
+[aws-badge]: https://img.shields.io/badge/AWS-Amazon_Web_Services-black?colorA=white&logoColor=232F3E&logo=amazonaws
+[aws]: https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication#amazon-web-services
+[azure-badge]: https://img.shields.io/badge/Azure-Microsoft_Azure-black?colorA=white&logoColor=0078D4&logo=microsoftazure
+[azure]: https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication#microsoft-azure
+[gcp-badge]: https://img.shields.io/badge/GCP-Google_Cloud_Platform-black?colorA=white&logoColor=4285F4&logo=googlecloud
+[gcp]: https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication#google-cloud-platform
+[k8s-badge]: https://img.shields.io/badge/K8s-Kubernetes-black?colorA=white&logoColor=326CE5&logo=kubernetes
+[k8s]: https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication#kubernetes
+[auth]: https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication
```
Comment (jorgeorpinel) [inline README.md:29]: 💅🏼 💅🏼 💅🏼  Shouldn't it be at the beginning of the link list? 😋

File: README.md [hunk 21]
```diff
@@ -1,25 +1,48 @@
...
+
+![](https://github.com/iterative/static/raw/main/img/tpi/high-level-light.png#gh-light-mode-only)
+![](https://github.com/iterative/static/raw/main/img/tpi/high-level-dark.png#gh-dark-mode-only)
```
Comment (jorgeorpinel) [inline README.md:32]: > `#gh-(light|dark)-mode-only`

Woah how does that work? Just curious

File: README.md [hunk 22]
```diff
@@ -1,25 +1,48 @@
...
+
+## What's Special
+
+There are a several reasons to use TPI instead of other related solutions (custom scripts and/or cloud orchestrators):
+
+1. **Reduced management overhead and infrastructure cost**:
+   TPI is a CLI tool, not a running service. It requires no additional orchestrating machine (control plane/head nodes) to schedule/recover/terminate instances. Instead, TPI runs (spot) instances via cloud-native scaling groups[^scalers], taking care of recovery and termination automatically on the cloud provider's side. This design reduces management overhead & infrastructure costs. You can close your laptop while cloud tasks are running -- auto-recovery happens even if you are offline.
```
Comment (jorgeorpinel) [inline README.md:39]: When we number the list it seems like we mean that it is an exhaustive list. Is it? Otherwise maybe use bullets.

But also it somewhat overlaps with the previous bullet list. Maybe they can be combined? So that the reader can get to the Usage right after the vendors figure

File: README.md [hunk 23]
```diff
@@ -92,14 +126,52 @@ TF_LOG_PROVIDER=INFO terraform refresh
 TF_LOG_PROVIDER=INFO terraform show
 ```
 
-### Stop Tasks
+### Stop Task
```
Comment (jorgeorpinel) [inline README.md:129]: End a Task?
End the Task?

File: README.md (hunk 14 above)
Comment (jorgeorpinel) [reply README.md:161]: Cool diagram but it's not readable without clicking the `<->` button (not necessarily obvious usability). Any chance to make it taller and less long i.e. fold the diagram somehow?

File: README.md [hunk 24]
```diff
@@ -92,14 +126,52 @@ TF_LOG_PROVIDER=INFO terraform refresh
...
+style C fill:#7B61FF,stroke:#333333,color:#000000
+```
+
+## Future Plans
+
+TPI is a CLI tool bringing the power of bare-metal cloud to a bare-metal local laptop. We're working on more featureful and visual interfaces. We'd also like to have more native support for distributed (multi-instance) training, more data sync optimisations & options, and tighter ecosystem integration with tools such as [DVC](https://dvc.org).
+
 ## Help
 
 The [getting started guide](https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/getting-started) has some more information. In case of errors, extra debugging information is available using `TF_LOG_PROVIDER=DEBUG` instead of `INFO`.
```
Comment (jorgeorpinel) [inline README.md:177]: 💅🏼 

```suggestion
The [Getting Started](https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/getting-started) guide has some more information. In case of errors, extra debugging information is available using `TF_LOG_PROVIDER=DEBUG` instead of `INFO`.
```

File: docs/guides/getting-started.md [hunk 25]
```diff
@@ -20,11 +20,11 @@ page_title: Getting Started
     sudo apt-get update && sudo apt-get install terraform
     ```
 
-- Create an account with any supported cloud vendor and expose its [authentication credentials via environment variables][authentication]
+- Create an account with any supported cloud vendor and expose its [authentication credentials via environment variables][auth]
 
-[authentication]: https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication
+[auth]: https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication
```
Comment (jorgeorpinel) [inline docs/guides/getting-started.md:25]: BTW I think the nav would be more logical with Getting Started on top, since it covers installation and then it links to the auth section.
